jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -301,20 +272,56 @@ void addScannerListener(Scanner.Listener listener)
         scanner.addListener(listener);
     }
 
+    protected int getDefaultWeight(String name)
+    {
+        URL url = getClass().getResource("environment-weights.properties");
```

Comment:
I don't think this will work if `DeploymentScanner` is subclassed.

Reply:
A properties file is too hard to change and it still makes weights in the "API".   We don't need weights, we just need an order.  If somebody defines a new environment, then setting a new order is easier than inventing a weight that is robust to future changes.